### PR TITLE
refactor: UPE banner content

### PR DESCRIPTION
**Pull request by frosso**
# Changes proposed in this Pull Request:

A few things about the UPE opt-in banner:
- The banner is now enqueued _only_ on the payment methods settings pages (and not on _all_ WC admin pages)
- The banner's content is now different between the settings pages and the main Stripe page (confirmed with UX)
- Adjusted the spacing and styles to more closely align to the design

Tested in Chrome, Safari, Firefox.

# Testing instructions
- Ensure the `_wcstripe_feature_upe_settings` flag is enabled
- Ensure the `woocommerce_stripe_settings.upe_checkout_experience_enabled` flag is disabled
- Go to the main Stripe page: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe
- The banner's content is now different:
Screen Shot 2021-09-09 at 5 08 48 PM
- On the other gateway pages, the banner is similar (but the content now dynamically includes the payment method's title)
Markup 2021-09-09 at 17 09 58
- Examples:
  - http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe_sepa
  - http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe_giropay
  - http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe_eps
  - http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe_bancontact
  - http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe_sofort
  - http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe_p24
  - http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe_alipay
  - http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe_multibanco
  - http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe_ideal

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -25,6 +25,6 @@ module.exports = {
 		upe_classic: './client/classic/upe/index.js',
 		upe_settings: './client/settings/index.js',
 		upe_onboarding_wizard: './client/upe-onboarding-wizard/index.js',
-		upe_opt_in_banner: './client/settings/upe-opt-in-banner/index.js',
+		upe_opt_in_banner: './client/entrypoints/upe-opt-in-banner/index.js',
```
Comment (frosso) [inline webpack.config.js:33]: Moved to a new `entrypoints` directory (for background pc4etw-jM-p2 )

File: includes/abstracts/abstract-wc-stripe-payment-gateway.php [hunk 2]
```diff
@@ -54,26 +54,48 @@ public function admin_options() {
...
+		global $current_tab, $current_section;
 
-			wp_enqueue_script( 'woocommerce_stripe_upe_opt_in' );
+		if ( ! isset( $current_tab ) || 'checkout' !== $current_tab ) {
+			return;
 		}
+
+		if ( ! isset( $current_section ) || $this->id !== $current_section ) {
+			return;
+		}
```
Comment (frosso) [inline includes/abstracts/abstract-wc-stripe-payment-gateway.php:73]: Now we're enqueuing the script/styles _onlny_ when necessary

File: client/settings/upe-opt-in-banner/index.js [hunk 3]
```diff
@@ -37,60 +49,45 @@ const Actions = styled.div`
...
 );
 
-const bannerContainer = document.getElementById(
-	'wc-stripe-upe-opt-in-banner'
-);
-
-if ( bannerContainer ) {
-	ReactDOM.render( <UpeOptInBanner />, bannerContainer );
-}
-
```
Comment (frosso) [inline client/settings/upe-opt-in-banner/index.js:95]: Before: importing the banner component _also_ had side effects on the DOM - now the banner component file doesn't have side effects

File: client/settings/upe-opt-in-banner/index.js [hunk 4]
```diff
@@ -11,23 +10,36 @@ import { Button, Card, ExternalLink } from '@wordpress/components';
  * Internal dependencies
  */
 import Pill from '../../components/pill';
-import AllPaymentMethodsIcon from '../../payment-method-icons/all-payment-methods';
+
+const StyledPill = styled( Pill )`
+	border-color: #007cba;
+	color: #007cba;
+`;
```
Comment (frosso) [inline client/settings/upe-opt-in-banner/index.js:17]: Color of the "early access" pill now more closely matches the designs

File: client/settings/general-settings-section/upe-opt-in-banner.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	/>
+);
```
Comment (frosso) [inline client/settings/general-settings-section/upe-opt-in-banner.js:20]: This is the same image that was in `client/payment-method-icons/all-payment-methods/icon.svg` - with a little less markup

**Comment by harriswong** [comment]
👀

**Review by harriswong — APPROVED** [review]
LGTM

- Tested `page=wc-settings&tab=checkout&section=stripe` and giropay. Verified the banners are different.

**Pull request merged**